src/sentry/notifications/utils/participants.py

```diff
@@ -383,10 +387,24 @@ def get_send_to(
     event: Event | None = None,
     notification_type: NotificationSettingTypes = NotificationSettingTypes.ISSUE_ALERTS,
     fallthrough_choice: FallthroughChoiceType | None = None,
+    rules: Iterable[Rule] | None = None,
 ) -> Mapping[ExternalProviders, set[RpcActor]]:
     recipients = determine_eligible_recipients(
         project, target_type, target_identifier, event, fallthrough_choice
     )
+
+    if rules:
+        rule_snoozes = RuleSnooze.objects.filter(Q(rule__in=rules))
+        muted_user_id = []
+        for rule_snooze in rule_snoozes:
+            if rule_snooze.user_id is None:
+                return {}
+            else:
+                muted_user_id.append(rule_snooze.user_id)
+
+        if muted_user_id:
+            recipients = list(filter(lambda x: x.id not in muted_user_id, recipients))
```

Comment:
Overall looking good :)

Reply:
I checked with Leander and it seems like id checking is ok? If I understood correctly, because it is just the id (and not anything else) , it's fine

Reply:
Hm I'm still skeptical here, I don't think a user id maps to an actor id. I pulled your branch and printed some stuff out, in `test_someone_else_snoozes_themself` `user2` doesn't even have an actor id.